erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -123,6 +125,29 @@ def validate_work_order_status(self):
 		if pro_doc.status == 'Completed':
 			frappe.throw(_("Cannot cancel transaction for Completed Work Order."))
 
+	def validate_items_qty_for_purchase_receipt(self):
+		pr_list = list(set([item.reference_purchase_receipt for item in self.items]))
+
+		already_transferred = frappe.get_all("Stock Entry Detail", filters=[('reference_purchase_receipt', 'in', pr_list),
+			('docstatus', '=', 1)], fields = ["item_code", "qty"])
+
+		max_item_to_transfer =  frappe.get_all("Purchase Receipt Item", filters=[("parent", "in", pr_list)], fields = ["parent", "item_code", "qty"])
+		max_item_to_transfer =  calculate_total_qty(max_item_to_transfer)
```

Comment:
Too many unnecessary calls, remove the method validate_items_qty_for_purchase_receipt.

Reply:
done